app/src/main/java/com/example/diceroller/DiceRollerViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.diceroller
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.getAndUpdate
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import java.util.Calendar
+import kotlin.random.Random
+
+private const val REPLAY_LAST: Int = 1
+
+class DiceRollerViewModel : ViewModel() {
+    private val currentDice: MutableStateFlow<DiceType> = MutableStateFlow(DiceType.D6)
+
+    private val _rollHistoryState: MutableStateFlow<List<Dice>> = MutableStateFlow(listOf())
+    val rollHistoryState: StateFlow<List<Dice>> get() = _rollHistoryState.asStateFlow()
+
+    private val _rollDieState: MutableSharedFlow<ViewState> = MutableSharedFlow(REPLAY_LAST)
+    val viewState: SharedFlow<ViewState> get() = _rollDieState.asSharedFlow()
```

Comment:
Let's talk about the pros and cons of a shared ViewModel. 

The good thing is that when several screens/views depend on the same view logic we can avoid being repetitive. 

The bad thing is that because two views need to read off the same State but at potentially different times, you need to add management and handling. This exposes you to complex things like `SharedFlow` which is fine, but also it tightly couples your views. If each view grow to have unrelated concerns you end up turning the ViewModel into spaghetti.